cirq/ops/common_channels.py

```diff
@@ -31,109 +31,196 @@
-        self._p_z = value.validate_probability(p_z, 'p_z')
-        self._p_i = 1 - value.validate_probability(p_x + p_y + p_z,
-                                                   'p_x + p_y + p_z')
+        if error_probabilities is not None:
+            for k in error_probabilities.keys():
+                if len(k) != num_qubits:
+                    raise ValueError('{} does not have {} entries'.format(
+                        k, num_qubits))
+            self._num_qubits = num_qubits
+            self._error_probabilities = error_probabilities
+        else:
+            if p_x is None:
+                p_x = 0.0
+            if p_y is None:
+                p_y = 0.0
+            if p_z is None:
+                p_z = 0.0
+
+            p_x = value.validate_probability(p_x, 'p_x')
+            p_y = value.validate_probability(p_y, 'p_y')
+            p_z = value.validate_probability(p_z, 'p_z')
+            p_i = 1 - value.validate_probability(p_x + p_y + p_z,
+                                                 'p_x + p_y + p_z')
+
+            self._num_qubits = 1
+            self._error_probabilities = {'I': p_i, 'X': p_x, 'Y': p_y, 'Z': p_z}
 
     def _mixture_(self) -> Sequence[Tuple[float, np.ndarray]]:
-        return ((self._p_i, protocols.unitary(identity.I)),
-                (self._p_x, protocols.unitary(pauli_gates.X)),
-                (self._p_y, protocols.unitary(pauli_gates.Y)),
-                (self._p_z, protocols.unitary(pauli_gates.Z)))
+        Pis = []
+        for pauli in self._error_probabilities.keys():
+            Pi = np.identity(1)
+            for gate in pauli:
+                if gate == 'I':
+                    Pi = np.kron(Pi, protocols.unitary(identity.I))
+                elif gate == 'X':
+                    Pi = np.kron(Pi, protocols.unitary(pauli_gates.X))
+                elif gate == 'Y':
+                    Pi = np.kron(Pi, protocols.unitary(pauli_gates.Y))
+                elif gate == 'Z':
+                    Pi = np.kron(Pi, protocols.unitary(pauli_gates.Z))
+            Pis.append(Pi)
+        return tuple(zip(self._error_probabilities.values(), Pis))
+
+    def _num_qubits_(self) -> int:
+        return self._num_qubits
 
     def _has_mixture_(self) -> bool:
         return True
 
     def _value_equality_values_(self):
-        return self._p_x, self._p_y, self._p_z
+        return self._num_qubits, hash(
+            tuple(sorted(self._error_probabilities.items())))
 
     def __repr__(self) -> str:
-        return 'cirq.asymmetric_depolarize(p_x={!r},p_y={!r},p_z={!r})'.format(
-            self._p_x, self._p_y, self._p_z
-        )
+        if self._num_qubits == 1:
+            return (
+                'cirq.asymmetric_depolarize(error_probabilities={!r})'.format(
+                    self._error_probabilities))
+        return ('cirq.asymmetric_depolarize' +
+                '(num_qubits={!r},error_probabilities={!r})'.format(
```

Comment:
We can actually simplify this since we don't need the `num_qubits` argument (see other comment).